armi/nuclearDataIO/cccc/nhflux.py

```diff
@@ -0,0 +1,477 @@
+        and DIF3D GEODST indexing. "External surfaces" are important, because they contain the
+        INCOMING partial currents from the outer reactor boundary.
+        This uses the same hex ordering as geodstCoordMap, except that each hex now has 6 subsequent indices.
+        If hex of index n (0 to N-1) has a surface of index k (0 to 5) that lies on the vacuum boundary,
+        then the index of that surface is N*6 + k + 1.
+
+    self.fluxMoments : 2-D list of float
+        This contains all the flux moments for all core assemblies at ONE axial position.
+        The jth planar flux moment of assembly i is fluxMoments[i][j].
+        The hex assemblies are ordered according to self.geodstCoordMap.
+
+    self.partialCurrentsHex : 2-D list of float
+        This contains all the OUTGOING partial currents for each assembly in the given axial plane.
+        The OUTGOING partial current on surface j in assembly i is surfCurrents[i][j].
+        The hex assemblies are ordered according to self.geodstCoordMap.
+
+    self.partialCurrentsHex_ext : 1-D list of floats
+        This contains all the INCOMING partial currents on "external hex surfaces", which are
+        adjacent to the reactor outer boundary (usually vacuum). Internal reflective surfaces
+        are NOT included in this!
+        These "external hex surfaces" are ordered according to self.externalCurrentPointers.
+
+    self.partialCurrentsZ  : 2-D list of float
+        This contains all the upward and downward partial currents in all assemblies on ONE whole-core axial slice.
+        The hex assemblies are ordered according to self.geodstCoordMap.
+
+    """
+
+    def __init__(self, fName="NHFLUX", variant=False, numDataSetsToRead=1):
+        """
+        Initialize the NHFLUX or NAFLUX reader object.
+
+        Parameters
+        ----------
+        fName : str, optional
+            The file name of the NHFLUX binary file to be read.
+
+        variant : bool, optional
+            Whether or not this NHFLUX/NAFLUX file has the VARIANT output format, which is a bit different than
+            the DIF3D nodal format.
+
+        """
+
+        cccc.DataContainer.__init__(self)
+
+        self.metadata["variantFlag"] = variant
+        self.metadata["numDataSetsToRead"] = numDataSetsToRead
+
+        # Initialize class array variables.
+        self.geodstCoordMap: numpy.ndarray = numpy.array([])
+        self.externalCurrentPointers: numpy.ndarray = numpy.array([])
+        self.fluxMoments: numpy.ndarray = numpy.array([])
+        self.partialCurrentsHex: numpy.ndarray = numpy.array([])
+        self.partialCurrentsHex_ext: numpy.ndarray = numpy.array([])
+        self.partialCurrentsZ: numpy.ndarray = numpy.array([])
+        self.incomingPointersToAllAssemblies: numpy.ndarray = numpy.array([])
+
+
+class NhfluxStream(cccc.StreamWithDataContainer):
+    @staticmethod
+    def _getDataContainer() -> NHFLUX:
+        return NHFLUX()
+
+    def readWrite(self):
```

Comment:
ugg and you cant make this private without having to re-do ISOTXS?

Reply:
this `readWrite` thing being public is one current bane of the cccc subpackags. It needs a new name and it needs to be private. It can't be `_readWrite` because we already have one. I was considering `_readWriteMain`, etc. for a while. 

Redoing isotxs won't be *too* hard.  It's pretty close.